### DISCUSSION TITLE
Way to stop a running scan on request

### DISCUSSION THREAD

**Discussion by pedrofialho94** [projectdiscovery/nuclei#5138] [Ideas]
It could be useful to have a way to stop a scan on request when using the Go library. Since both functions currently used are blocking and do not accept any signals from the outside, I don't think this is possible to implement at the moment.

**Comment by GeorginaReeder** [comment]
Hey @pedrofialho94 ! Thanks for starting this discussion. Could you go ahead and create a feature request for this? We'd be happy to take a look from there.

**Comment by ehsandeep** [comment]
@pedrofialho94 see https://github.com/projectdiscovery/nuclei/pull/5096

**Reply by pedrofialho94** [reply]
@ehsandeep thanks for letting me know about this. If I'm looking at the code correctly, there's currently no way for a user of the sdk to send their own context into the engine, right?

Would it be interesting to create a feature request for this?

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
